File: Data Structures & Algorithms/reverse-a-linked-list/submission-0.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        val = []
        while head:
            val.append(head.val);
            head = head.next;
        if len(val) == 0:
            return None
        root = ListNode(val.pop())
        cur = root
        while len(val):
            cur.next = ListNode(val.pop())
            cur = cur.next
        return root
        

File: Data Structures & Algorithms/count-paths/submission-0.cpp
class Solution {
public:
    int uniquePaths(int m, int n) {
        int d[101][101] = {0};
        d[1][1] = 1;
        for(int i = 1; i <= m; ++i) {
            for(int j = 1; j <= n; ++j)
                d[i][j] += d[i-1][j] + d[i][j-1];
        }
        return d[m][n];
    }
};
